AGENTS.md

```diff
@@ -0,0 +1,44 @@
+# Repository Guidelines for Codex
+
+The instructions in this document apply to the entire repository.
+
+## Build & Test
+
+- Check typings with `deno task check`.
+- Run all tests using `deno task test-all`.
+- To run a single test file use `deno test path/to/test.ts`.
+- To test a specific package, `cd` into the package directory and run
+  `deno task test`.
+
+## Formatting
+
+- Line width is **80 characters**.
+- Indent with **2 spaces**.
+- **Semicolons are required.**
+- Use **double quotes** for strings.
+- Always run `deno fmt` before committing.
+
+## TypeScript
+
+- Export types explicitly using `export type { ... }`.
+- Provide descriptive JSDoc comments on public interfaces.
```

Comment:
perhaps we should add something that we should check affected packages README.md to update them?

Reply:
might be worth experimenting with this